packages/server/lib/project-base.ts

```diff
@@ -624,7 +619,11 @@ export class ProjectBase extends EE {
     const isDefaultProtocolEnabled = this._protocolManager?.isProtocolEnabled ?? false
 
     // hide the runner if explicitly requested or if the protocol is enabled outside of studio and the runner is not explicitly enabled
-    const hideRunnerUi = this.options?.args?.runnerUi === false || (isDefaultProtocolEnabled && !this.ctx.coreData.studio && !this.options?.args?.runnerUi)
+    const studio = this.ctx.coreData.studioLifecycleManager?.getStudioIfReady()
+    const hideRunnerUi = (
+      this.options?.args?.runnerUi === false ||
+      (isDefaultProtocolEnabled && !studio && !this.options?.args?.runnerUi)
+    )
```

Comment:
Thinking through this, I think we can remove studio from the equation entirely (which would mean we no longer need the code on lines 179-185 and no longer need `getStudioIfReady` as a function at all.

We should change checking for studio to checking if we're in run mode:

```suggestion
    const studio = this.ctx.coreData.studioLifecycleManager?.getStudioIfReady()
    const hideRunnerUi = (
      this.options?.args?.runnerUi === false ||
      (isDefaultProtocolEnabled && !this._cng.isTextTerminal && !this.options?.args?.runnerUi)
    )
```